controller/metrics/metrics.go

```diff
@@ -193,7 +193,7 @@ func NewMetricsServer(addr string, appLister applister.ApplicationLister, appFil
 		redisRequestCounter:     redisRequestCounter,
 		redisRequestHistogram:   redisRequestHistogram,
 		hostname:                hostname,
-		cron:                    cron.New(),
+		cron:                    cron.New(cron.WithParser(cron.NewParser(cron.SecondOptional | cron.Minute | cron.Hour | cron.Dom | cron.Month | cron.Dow | cron.Descriptor))),
```

Comment:
I'm not sure we actually need to set a parser here. As far as I can tell, we don't use a parser in the metrics server.

But I think we might need something like this:

```go
cron.New(cron.WithChain(
    cron.Recover(logger), // or use cron.DefaultLogger
))
```

Since the new behavior is to _not_ recover panics in cron jobs, I want to make sure we match the old behavior.

Reply:
This is the place where the cron is actually uses:

```go
	err := m.cron.AddFunc(fmt.Sprintf("@every %s", cacheExpiration), func() {
		log.Infof("Reset Prometheus metrics based on existing expiration '%v'", cacheExpiration)
		m.syncCounter.Reset()
		m.kubectlExecCounter.Reset()
		m.kubectlExecPendingGauge.Reset()
		m.k8sRequestCounter.Reset()
		m.clusterEventsCounter.Reset()
		m.redisRequestCounter.Reset()
		m.reconcileHistogram.Reset()
		m.redisRequestHistogram.Reset()
	})
```

So in robfig/cron v3, a panic from the log statement or any of the resets would not be recovered by the cron library.